src/els_dt_documents.erl

```diff
@@ -0,0 +1,83 @@
+%%==============================================================================
+%% The 'documents' table
+%%==============================================================================
+
+-module(els_dt_documents).
+
+%%==============================================================================
+%% Behaviour els_db_table
+%%==============================================================================
+
+-export([ name/0
+        , opts/0
+        ]).
+
+%%==============================================================================
+%% API
+%%==============================================================================
+
+-export([ insert/1
+        , lookup/1
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% Item Definition
+%%==============================================================================
+
+-record(els_dt_documents, { uri      :: uri()  | '_'
+                          , document :: els_document:document()
+                          }).
+-type els_dt_documents() :: #els_dt_documents{}.
+
+-type item() :: #{ uri      := uri()
+                 , document := els_document:document()
+                 }.
+-export_type([ item/0 ]).
+
+%%==============================================================================
+%% Callbacks for the els_db_table Behaviour
+%%==============================================================================
+
+-spec name() -> atom().
+name() -> ?MODULE.
+
+-spec opts() -> proplists:proplist().
+opts() ->
+  [ {attributes , record_info(fields, els_dt_documents)}
+  , {disc_copies, [node()]}
+  , {index      , []}
+  , {type       , set}
+  ].
+
+%%==============================================================================
+%% API
+%%==============================================================================
+
+-spec from_map(item()) -> els_dt_documents().
+from_map(#{ uri := Uri, document := Document }) ->
+  #els_dt_documents{ uri = Uri, document = Document }.
+
+-spec to_map(els_dt_documents()) -> item().
+to_map(#els_dt_documents{ uri = Uri, document = Document }) ->
```

Comment:
This could be renamed to `to_item` if the suggestion above is applied.

Reply:
Sounds reasonable. Will do.